addons/stock/models/stock_move.py

```diff
@@ -516,31 +511,25 @@ def write(self, vals):
                 receipt_moves_to_reassign |= move_to_unreserve.filtered(lambda m: m.location_id.usage == 'supplier')
                 receipt_moves_to_reassign |= (self - move_to_unreserve).filtered(lambda m: m.location_id.usage == 'supplier' and m.state in ('partially_available', 'assigned'))
 
-        # Handle the propagation of `date_expected` and `date` fields.
+        # Handle the propagation of `date` fields.
         propagated_date_field = False
-        if vals.get('date_expected'):
-            propagated_date_field = 'date_expected'
-        elif vals.get('state', '') == 'done' and vals.get('date'):
+        if vals.get('date'):
             propagated_date_field = 'date'
```

Comment:
propagated_date_field is useless now

Reply:
It is clean later by f492e51d732ad433226afc56753d47a55ff79c95